src/envelope.rs

```diff
@@ -78,12 +115,36 @@ impl<A: Handler<M, Return = R>, M: Send + 'static, R: Send + 'static> MessageEnv
         act: &'a mut Self::Actor,
         ctx: &'a mut Context<Self::Actor>,
     ) -> BoxFuture<'a, ()> {
+        #[cfg(feature = "with-tracing-0_1")]
         let Self {
             message,
             result_sender,
+            instrumentation,
             ..
         } = *self;
-        Box::pin(act.handle(message, ctx).map(move |r| {
+
+        #[cfg(not(feature = "with-tracing-0_1"))]
+        let Self {
+            message,
+            result_sender,
+            ..
+        } = *self;
+
+        let fut = act.handle(message, ctx);
+
+        #[cfg(feature = "with-tracing-0_1")]
+        let fut = {
+            let Instrumentation { parent, .. } = instrumentation;
+            let executing = debug_span!(
+                parent: &parent,
+                "xtra message handler",
+                actor = std::any::type_name::<A>(),
+                message = std::any::type_name::<M>(),
+            );
+            fut.instrument(executing)
+        };
```

Comment:
If we would delegate this to `Instrumentation`, like:

```rust
let fut = instrumentation.apply(fut);
```

Then we could also do this unconditionally and simply have a no-op impl for when the feature is not enabled.

Reply:
This would also reduce code duplication between the various envelopes.